### PR TITLE
Fix mxnet ctc_loss bug

### PR DESCRIPTION
**Pull request by HawkAaron**
1. fix ctc_loss GPU bug
~2. add blank label for gluon CTCLoss~
edit @szha: crossed out second item

### CODE REVIEW

File: python/mxnet/gluon/loss.py [hunk 1]
```diff
@@ -409,6 +409,8 @@ class CTCLoss(Loss):
         length respectively.
     weight : float or None
         Global scalar weight for loss.
+    blank_label : {'first', 'last'}, default 'last'
```
Comment (szha) [inline python/mxnet/gluon/loss.py]: it was intentional not to expose this option in gluon.

**Comment by chinakook** [comment]
Is the speed of big vocab size solved of this op?

File: python/mxnet/gluon/loss.py (hunk 1 above)
Comment (HawkAaron) [reply python/mxnet/gluon/loss.py]: That means I need to revert this commit and  resend a pull request ?

**Comment by HawkAaron** [comment]
@chinakook its looks good to me

File: python/mxnet/gluon/loss.py (hunk 1 above)
Comment (szha) [reply python/mxnet/gluon/loss.py]: you can simply add a commit that removes the change in blank_label

**Comment by szha** [comment]
@Jerryzcn could you give this a try?

**Comment by Jerryzcn** [comment]
will do

**Comment by Jerryzcn** [comment]
@HawkAaron it seems removing the max negatively affect convergence. Do you observe similar result?

File: src/operator/contrib/ctc_include/detail/gpu_ctc.h [hunk 2]
```diff
@@ -411,12 +411,7 @@ GpuCTC<ProbT>::compute_log_probs(const ProbT* const activations) {
         denoms_, out_dim_, num_elements);
 
     // compute denominators for softmax
-    denoms_handle = reduce_with_axis<red::sum, false>(
-        F<mxnet::op::mshadow_op::exp>(
-            log_probs_handle -
-            broadcast<0>(reduce_with_axis<red::maximum, false>(log_probs_handle, 1),
```
Comment (Jerryzcn) [inline src/operator/contrib/ctc_include/detail/gpu_ctc.h]: max is necessary here.

Comment (HawkAaron) [reply src/operator/contrib/ctc_include/detail/gpu_ctc.h]: the max is reduced in here: https://github.com/apache/incubator-mxnet/blob/master/src/operator/contrib/ctc_include/detail/gpu_ctc.h#L398

and `log_probs -= denoms` is in here: https://github.com/apache/incubator-mxnet/blob/master/src/operator/contrib/ctc_include/detail/gpu_ctc.h#L409

that means the broadcast line (https://github.com/apache/incubator-mxnet/blob/master/src/operator/contrib/ctc_include/detail/gpu_ctc.h#L417) should be zero

Comment (Jerryzcn) [reply src/operator/contrib/ctc_include/detail/gpu_ctc.h]: I see. Did not look at other parts of the code thanks! Let me check why my model is not converging

**Comment by szha** [comment]
@HawkAaron thanks for the fix, and @Jerryzcn thanks for the review

**Pull request merged**